Updated documentation to use range notation syntax

genbattle opened the pull request:
Replaced outdated use of the `range(start, end)` function where appropriate with `start..end`, and tweaked the examples to compile and run with the latest rust. I also fixed two periphery compile issues in reference.md which were occluding whether there were any new errors created by these changes.

rust-highfive commented:
Thanks for the pull request, and welcome! The Rust team is excited to review your changes, and you should hear from @steveklabnik (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way Github handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTING.md](https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md) for more information.

edwardw commented on src/doc/reference.md:
```diff
@@ -2994,7 +2994,7 @@ Some examples of call expressions:
 
 let x: i32 = add(1i32, 2i32);
-let pi: Option<f32> = "3.14".parse().ok();
+let pi: Option<f32> = "3.14".parse();
```
`let pi: Result<f32, _> = "3.14".parse();` The API has been changed.

genbattle replied:
When was it updated? [The documentation](http://doc.rust-lang.org/std/str/trait.StrExt.html#tymethod.parse) still seems to refer to it returning an Option<F>.

edwardw replied:
It is #21718, just yesterday fresh out of the oven.

genbattle replied:
I have no compiler to test against, the latest rust-nightly-bin package is (1d00c545e 2015-01-30 19:56:34 +0000). I can't fix this issue and verify that it works until the nightly is updated to include your change; currently I just get:
```
main.rs:12:13: 12:27 error: wrong number of type arguments: expected 1, found 2 [E0244]
main.rs:12     let pi: Option<f32, _> = "3.14".parse();
```

steveklabnik replied:
You have `Option<` in that error message, not `Result<`.

genbattle replied:
Why not open a separate PR to update all the doc examples affected by #21718 instead of cramming it into this one?

edwardw replied:
That's, of course, a viable solution too. Fine with me.

steveklabnik commented:
This needs a rebase now, but I'm interested in moving it forward.

genbattle commented:
I know how to rebase my branch against master in my fork, but I haven't used github enough to figure out the best way to update my fork with mainline rust - github seems to suggest pulling the changes from mainline rust via a pull request into my fork. Is this the best way to go about it, or is there a cleaner/easier way to do this?

genbattle commented:
No matter, figured it out, rebasing now.

steveklabnik commented:
I have this setup:

```
$ git remote -v
origin  git@github.com:steveklabnik/rust.git (fetch)
origin  git@github.com:steveklabnik/rust.git (push)
upstream        https://github.com/rust-lang/rust.git (fetch)
upstream        https://github.com/rust-lang/rust.git (push)
```

and then

```
$ git fetch upstream
$ git rebase upstream/master # with all this entails...
$ git push origin -f mybranch
```

genbattle commented:
After I did the rebase git insisted I pull and merge with my remote, but obviously in this case a push -f would have been the better option. It seems to have worked, although I don't know what it's done to the history :-/

fujin commented:
Use the reflog to hard move your branch pointer back before you pull/merged, rebase again, check fast-forward, then `git push --force-with-lease`. ~0~

genbattle commented:
I reset to before the pull/merge and force pushed, it seems to have corrected it?

fujin commented:
@genbattle Indeed; Thank you.

genbattle commented:
Steve, don't push this through yet. I just realized I don't think I ended
up fixing up newly created occurrences of range(start, end) when I did the
merge for the rebase, I'll get onto this in the morning and add another
commit to this PR.
On 6 Feb 2015 6:49 pm, "AJ Christensen" <notifications@github.com> wrote:

> @genbattle <https://github.com/genbattle> Indeed; Thank you.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rust-lang/rust/pull/21834#issuecomment-73186168>.
>

genbattle commented:
There was only one additional instance of the range function instead of range notation, so I've fixed it up, tested it, and added it to this branch.

steveklabnik commented:
This looks great, @genbattle , except the original comment about `Result`. Do these tests pass? Other than that, this looks great, r=me

genbattle commented:
@steveklabnik I've fixed the usage of StrExt.parse() that @edwardw originally commented on. I also fixed all similar occurrences in doc/trpl/guessing-game.md.

I did another rebase and tested against all of the examples in all of the files touched by this PR and fixed any errors that arose. It should be all good to go unless you have any more comments or recommendations for me?

steveklabnik commented:
@bors: r+ 8300095

steveklabnik commented:
Nope, looks great! Thnak you! (not rolling up since this is a big ish change)

bors commented:
:hourglass: Testing commit 8300095 with merge 4760750...

steveklabnik commented:
@bors: retry

bors commented:
:hourglass: Testing commit 8300095 with merge 5b543c1...

bors commented:
:broken_heart: Test failed - [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/3598)

alexcrichton commented:
@bors: retry

bors commented:
:hourglass: Testing commit 8300095 with merge f174bca...

bors commented:
:sunny: Test successful - [auto-linux-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-32-nopt-t/builds/3594), [auto-linux-32-opt](http://buildbot.rust-lang.org/builders/auto-linux-32-opt/builds/3598), [auto-linux-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-linux-64-nopt-t/builds/3589), [auto-linux-64-opt](http://buildbot.rust-lang.org/builders/auto-linux-64-opt/builds/3597), [auto-linux-64-x-android-t](http://buildbot.rust-lang.org/builders/auto-linux-64-x-android-t/builds/3572), [auto-mac-32-opt](http://buildbot.rust-lang.org/builders/auto-mac-32-opt/builds/3609), [auto-mac-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-mac-64-nopt-t/builds/3614), [auto-mac-64-opt](http://buildbot.rust-lang.org/builders/auto-mac-64-opt/builds/3606), [auto-win-32-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-32-nopt-t/builds/3266), [auto-win-32-opt](http://buildbot.rust-lang.org/builders/auto-win-32-opt/builds/3261), [auto-win-64-nopt-t](http://buildbot.rust-lang.org/builders/auto-win-64-nopt-t/builds/2091), [auto-win-64-opt](http://buildbot.rust-lang.org/builders/auto-win-64-opt/builds/2099)
